extensions/ql-vscode/src/databases/github-databases/api.ts

```diff
@@ -68,52 +65,21 @@ export async function listDatabases(
   credentials: Credentials,
   config: GitHubDatabaseConfig,
 ): Promise<ListDatabasesResult | undefined> {
-  // On GHEC-DR, unauthenticated requests will never work, so we should always ask
-  // for authentication.
-  const hasAccessToken =
-    !!(await credentials.getExistingAccessToken()) || hasGhecDrUri();
+  const hasAccessToken = !!(await credentials.getExistingAccessToken());
 
-  let octokit = hasAccessToken
-    ? await credentials.getOctokit()
-    : new AppOctokit();
-
-  let promptedForCredentials = false;
-
-  let databases: CodeqlDatabase[];
-  try {
-    const response = await octokit.rest.codeScanning.listCodeqlDatabases({
-      owner,
-      repo,
-    });
-    databases = response.data;
-  } catch (e) {
-    // If we get a 404 when we don't have an access token, it might be because
-    // the repository is private/internal. Therefore, we should ask the user
-    // whether they want to connect to GitHub and try again.
-    if (e instanceof RequestError && e.status === 404 && !hasAccessToken) {
-      // Check whether the user wants to connect to GitHub
-      if (!(await askForGitHubConnect(config))) {
-        return;
-      }
-
-      // Prompt for credentials
-      octokit = await credentials.getOctokit();
-
-      promptedForCredentials = true;
-
-      const response = await octokit.rest.codeScanning.listCodeqlDatabases({
-        owner,
-        repo,
-      });
-      databases = response.data;
-    } else {
-      throw e;
-    }
+  if (!hasAccessToken && !(await askForGitHubConnect(config))) {
```

Comment:
The `askForGitHubConnect` function allows you to dismiss a dialog as `Never`, which means that github will not try to download a database.

Previously, it looks like `askForGitHubConnect` was only called if there was no token available. Now it looks like it is always called.

Does this mean that if `Never` is selected, then databases can't be downloaded even if a user explicitly requests it? I'm not exactly sure of the flow, so I may be missing something.